### PR TITLE
kubelet_stats: fix potential e2e crash dereferencing ContainerStats.CPU

### PR DESCRIPTION
**Pull request by rphillips**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
The kubelet's stats API can return nil within the CPU attribute of cadvisorapiv2.ContainerStats. This patch fixes the e2e test to check for nil CPU attributes before trying to dereference `oldStats.CPU` and `cStats.CPU`. We are seeing this flake currently within CI.

ref: https://bugzilla.redhat.com/show_bug.cgi?id=1679612
ref: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/stats/helper.go#L36-L40
ref: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/stats/helper.go#L77-L84

/cc @kubernetes/sig-node-pr-reviews @sjenning 

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Review by mattjmcnaughton — APPROVED** [review]
Thanks for this fix :)

/lgtm
/assign @BenTheElder

**Review by mattjmcnaughton — CHANGES_REQUESTED** [review]
/lgtm cancel

whoops, jumped the gun here.  I'm not sure this is actually going to work the way we are hoping.

I think this change will just push the nil failure back to the `computeContainerResourceUsage` method.

Maybe we should also `continue` if either `cStats` or `oldStats` is nil?

**Comment by rphillips** [comment]
@mattjmcnaughton thanks for the review... I've tweaked the PR with your suggestion.

**Review by mattjmcnaughton — COMMENTED** [review]
Cool, I'm convinced that this won't crash :)

My only request is for someone else with more context on this code than me to confirm that we *do not* want to treat either `oldStats` or `cStats` being new as a test failure. I'll let whoever does that confirmation approve.

File: test/e2e/framework/kubelet_stats.go
Comment (RobertKrawitz) [inline test/e2e/framework/kubelet_stats.go]: cStats.Memory looks like it might be subject to the same problem if that comes back nil.

Comment (rphillips) [reply test/e2e/framework/kubelet_stats.go]: Thanks for the review... Updated the PR :)

**Review by BenTheElder — APPROVED** [review]
/approve
/hold
this seems reasonable, holding in case we still want input from @timothysc, feel free to hold cancel to unflake and follow up later :-)

**Comment by timothysc** [comment]
I don't see an issue, but I'd defer to @kubernetes/sig-node-pr-reviews for final lgtm.

**Pull request merged**
